eesp-ikev2.org

```diff
@@ -242,76 +165,117 @@ in high-throughput or low-latency environments, where rapid setup and
-numbers are in use.  For a given Sub SA key, sequence numbers MUST
+numbers are in use. For a given Sub SA key, sequence numbers MUST
 remain unique and monotonically increasing to meet cryptographic
 requirements.
 
-** Negotiation of Sub SA
+** EESP Sequence Numbers
 
-#+caption: Sub SA Notifier
-#+name: sub-sa-notifier
-#+begin_src
-                    1                   2                   3
-0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-+-+-----------------------------+-------------------------------+
-! Next Payload  !C!  RESERVED   !         Payload Length        !
-+---------------+---------------+-------------------------------+
-!  Protocol ID  !   SPI Size    !      Notify Message Type      !
-+---------------+---------------+-------------------------------+
-!  Minimum number of Sub SAs    |      RESERVED                 !
-+-------------------------------+-------------------------------+
-#+end_src
+Unlike ESP, EESPv0 header allows to transmit 64-bit sequence numbers.
+In addition, the Sequence Number field in the EESPv0 header is optional
+and can be omitted from the packet if replay protection service is 
+not needed.
 
-- Protocol ID (1 octet) - MUST be 0. MUST be ignored if not 0.
-- SPI Size (1 octet) - MUST be 0. MUST be ignored if not 0.
-- Notify Status Message Type (2 octets) - set to <TBDXXX>.
-- Minimum number of Sub SAs (2 octets). MUST be greater than 0.
-  If 0 is received, it MUST be interpreted as 1.
+# EESP provides an optional Replay service using
+# 64-bit sequence numbers carried in the packet.
+
+# [VS] This should go to the Security Considerations
+Because the Replay Protection service is disabled, an attacker can re
+play packets with a different source address. Such an attacker could
+disrupt the connection by replaying a single packet with a different
+source address or port number.
 
-** UDP Encapsulation of Sub SA
+** Explicit Initialization Vector
+
+If the algorithm used to encrypt the payload requires cryptographic
+synchronization data, e.g., an Initialization Vector (IV), then this
+may be carried explicitly in every EESP packet.
+
+** Implicit Initialization Vectors
+
+With the Implicit Initialization Vector (IIV) encryption algorithm,
+as specified in [[RFC8750]], the IV MUST be omitted in the EESP
+packet. To enable this functionality, IIV transforms defined in
+[[IKEv2-Enc]] MUST be used during negotiation. Furthermore,
+the [[IKEv2-SN]] extension MUST be negotiated to support the use of
+64-bit Sequential Numbers in EESP packets. If the the proposal
+does not include 64-bit Sequential Numbers return error
+NO_PROPOSAL_CHOSEN.
+
+** Session ID
+
+The Session ID is a multi-purpose attribute with mutually
+exclusive values. The initiator MUST propose a single value in the
+Child SA proposal, Transform EESPSSID (Value). The responder MUST
+either accept the proposed value or reject it with an
+INVALID_SESSION_ID error message, indicating a supported value.
```

Comment:
Hm, not sure about that negotiation. That seems to violate the Sub SA use case (or would the value be combined with the Sub SA ID?). And as discussed before, using transforms is also not ideal for this.

Reply:
This text is a leftover. It must be edited.